tests/baselines/reference/declarationEmitAliasInlineing.js

```diff
@@ -0,0 +1,85 @@
+//// [tests/cases/compiler/declarationEmitAliasInlineing.ts] ////
+
+//// [a.ts]
+type O = {
+    prop: string
+    prop2: string
+}
+
+type I = {
+    prop: string
+}
+
+export const fn = (v: O['prop'], p: Omit<O, 'prop'>, key: keyof O, p2: Omit<O, keyof I>) => {};
+
+//// [aExp.ts]
+export type O = {
+    prop: string
+    prop2: string
+}
+
+export type I = {
+    prop: string
+}
+
+export const fnExp = (v: O['prop'], p: Omit<O, 'prop'>, key: keyof O, p2: Omit<O, keyof I>) => {};
+
+//// [b.ts]
+import {fn} from './a'
+import {fnExp} from './aExp'
+export const f = fn;
+export const fExp = fnExp;
+
+//// [a.js]
+"use strict";
+Object.defineProperty(exports, "__esModule", { value: true });
+exports.fn = void 0;
+var fn = function (v, p, key, p2) { };
+exports.fn = fn;
+//// [aExp.js]
+"use strict";
+Object.defineProperty(exports, "__esModule", { value: true });
+exports.fnExp = void 0;
+var fnExp = function (v, p, key, p2) { };
+exports.fnExp = fnExp;
+//// [b.js]
+"use strict";
+Object.defineProperty(exports, "__esModule", { value: true });
+exports.fExp = exports.f = void 0;
+var a_1 = require("./a");
+var aExp_1 = require("./aExp");
+exports.f = a_1.fn;
+exports.fExp = aExp_1.fnExp;
+
+
+//// [a.d.ts]
+type O = {
+    prop: string;
+    prop2: string;
+};
+type I = {
+    prop: string;
+};
+export declare const fn: (v: O["prop"], p: Omit<O, "prop">, key: keyof O, p2: Omit<O, keyof I>) => void;
+export {};
+//// [aExp.d.ts]
+export type O = {
+    prop: string;
+    prop2: string;
+};
+export type I = {
+    prop: string;
+};
+export declare const fnExp: (v: O["prop"], p: Omit<O, "prop">, key: keyof O, p2: Omit<O, keyof I>) => void;
+//// [b.d.ts]
+export declare const f: (v: string, p: Omit<{

```

Comment:
I'm a bit surprised we don't emit this as
```ts
export declare const f: typeof import("./a").fn;
```
since we're willing to emit the `import("./aExp").O["prop"]` et al.

Reply:
`typeof` is currently only used for function declarations. I think leaning into `typeof` for variables is an options that should be considered. I'm not sure why only function declarations use `typeof` (although I haven't really though too deeply about it)